pulser-core/pulser/sampler/samples.py

```diff
@@ -3,10 +3,11 @@
 
 from collections import defaultdict
 from dataclasses import dataclass, field, replace
-from typing import TYPE_CHECKING, Optional, cast
+from typing import TYPE_CHECKING, Any, Optional, cast
 
 import numpy as np
 
+# from pulser.sequence._schedule import _TimeSlot
```

Comment:
As you are only using it for type checking you can move the import to `if TYPE_CHECKING`. This will solve the circular import and one of your mypy issues.